apps/air-discount-scheme/backend/src/app/modules/flight/flight.controller.ts

```diff
@@ -86,23 +223,63 @@ export class PublicFlightController {
       throw new ForbiddenException('User postalcode does not meet conditions')
     }
 
-    const {
-      unused: flightLegsLeft,
-    } = await this.flightService.countThisYearsFlightLegsByNationalId(
-      discount.nationalId,
-    )
-    if (flightLegsLeft < flight.flightLegs.length) {
-      throw new ForbiddenException('Flight leg quota is exceeded')
+    let connectingFlight = false
+    let connectingId = undefined
+
+    let hasReykjavik = false
+    let hasAkureyri = false
+    for (const flightLeg of flight.flightLegs) {
+      if (
+        REYKJAVIK_FLIGHT_CODES.includes(flightLeg.origin) ||
+        REYKJAVIK_FLIGHT_CODES.includes(flightLeg.destination)
+      ) {
+        hasReykjavik = true
+      }
+      if (
+        AKUREYRI_FLIGHT_CODES.includes(flightLeg.origin) ||
+        AKUREYRI_FLIGHT_CODES.includes(flightLeg.destination)
+      ) {
+        hasAkureyri = true
+      }
     }
+
+    if (!hasReykjavik) {
+      connectingFlight = await this.validateConnectionFlights(
+        discount,
+        params.discountCode,
+        flight.flightLegs as FlightLeg[],
+      )
+      if (connectingFlight) {
```

Comment:
You can simplify this if `this.validateConnectionFlights` would return `connectingId` instead of a boolean

Reply:
A good point :+1: